streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -417,11 +277,11 @@ public void shouldSetRecordMetadata() {
 
         testDriver.pipeInput(consumerRecord1);
 
-        final List<Record> processedRecords = mockProcessors.get(0).processedRecords;
+        final List<String> processedRecords = supplier.theCapturedProcessor().processed;
         assertEquals(1, processedRecords.size());
 
-        final Record record = processedRecords.get(0);
-        final Record expectedResult = new Record(consumerRecord1, 0L);
+        final String record = processedRecords.get(0);
+        final String expectedResult = MockProcessor.makeRecord(consumerRecord1.key(), consumerRecord1.value(), consumerRecord1.timestamp());
```

Comment:
nit: import `makeRecord` ?

Reply:
Could we make `makeRecord()` more flexible and change it's parameters list to accept a `ConsumerRecord` ?